microsoft/testsuites/vm_extensions/runtime_extensions/common.py

```diff
@@ -108,9 +109,20 @@ def retrieve_storage_blob_url(
     blob = container_client.get_blob_client(blob_name)
     if not blob.exists():
         if is_public_container:
-            container_client.set_container_access_policy(
-                signed_identifiers={}, public_access="container"
-            )
+            try:
+                container_client.set_container_access_policy(
+                    signed_identifiers={}, public_access="container"
+                )
+            except ResourceExistsError as ex:
+                if (
+                    "public access is not permitted on this storage account"
+                    in str(ex).lower()
+                ):
+                    raise SkippedException(
+                        "Public access is not permitted on this storage account "
+                        f"{storage_account_name}, exception is '{ex}'"
```

Comment:
Don't use quotes on previous exception message, it may not be good on format.

```suggestion
                        f"{storage_account_name}. {ex}"
```

Reply:
updated